src/ImpactX.cpp

```diff
@@ -199,6 +215,95 @@ namespace impactx {
                                        << " slice_step=" << slice_step << "\n";
                     }
 
+                    //CSR Wakefield Response
+
+                    bool element_has_csr = false; // Updates to true for example with bend element
+                    amrex::Real R = 0.0; // Updates for bend element rc
+
+                    //Define lambda function inside of std::visit
+                    std::visit([&R, &element_has_csr](auto &&element)
+                    {
+                        if constexpr (std::is_same_v<std::decay_t<decltype(element)>, Sbend>)
+                        {
+                            R = element.m_rc;
+                            R = std::abs(R);
+                            element_has_csr = true;
+                        }
+                        else if constexpr (std::is_same_v<std::decay_t<decltype(element)>, CFbend>)
+                        {
+                            R = element.m_rc;
+                            R = std::abs(R);
+                            element_has_csr = true;
+                        }
+
```

Comment:
All these lines I would put inside a function call, so that `ImpactX.cpp` stays focused on overall control logic, and this then is called in a CSR-effect function that checks if CSR shall be calculated for an element and how.